soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1308,127 +1759,31 @@ s16 Randomizer::GetItemFromGet(RandomizerGet randoGet, GetItemID ogItemId) {
                 case 2:
                     return GI_GAUNTLETS_GOLD;
             }
-            return GI_RUPEE_BLUE;
-        
-        case RG_PROGRESSIVE_BOMB_BAG:
-            switch (CUR_UPG_VALUE(UPG_BOMB_BAG)) {
-                case 0:
-                    return GI_BOMB_BAG_20;
-                case 1:
-                    return GI_BOMB_BAG_30;
-                case 2:
-                    return GI_BOMB_BAG_40;
-            }
-            return GI_RUPEE_BLUE;
-
-        case RG_PROGRESSIVE_BOW:
-            switch (CUR_UPG_VALUE(UPG_QUIVER)) {
-                case 0:
-                    return GI_BOW;
-                case 1:
-                    return GI_QUIVER_40;
-                case 2:
-                    return GI_QUIVER_50;
-            }
-            return GI_RUPEE_BLUE;
-
-        case RG_PROGRESSIVE_SLINGSHOT:
-            switch (CUR_UPG_VALUE(UPG_BULLET_BAG)) {
-                case 0:
-                    return GI_SLINGSHOT;
-                case 1:
-                    return GI_BULLET_BAG_40;
-                case 2:
-                    return GI_BULLET_BAG_50;
-            }
-            return GI_RUPEE_BLUE;
-
         case RG_PROGRESSIVE_WALLET:
             switch (CUR_UPG_VALUE(UPG_WALLET)) {
                 case 0:
                     return GI_WALLET_ADULT;
                 case 1:
                     return GI_WALLET_GIANT;
             }
-            return GI_RUPEE_BLUE;
-
         case RG_PROGRESSIVE_SCALE:
             switch (CUR_UPG_VALUE(UPG_SCALE)) {
                 case 0:
                     return GI_SCALE_SILVER;
                 case 1:
                     return GI_SCALE_GOLD;
             }
-            return GI_RUPEE_BLUE;
-
-        case RG_PROGRESSIVE_NUT_UPGRADE:
-            switch (CUR_UPG_VALUE(UPG_NUTS)) {
-                case 0:
-                case 1:
-                    return GI_NUT_UPGRADE_30;
-                case 2:
-                    return GI_NUT_UPGRADE_40;
-            }
-            return GI_RUPEE_BLUE;
-
-        case RG_PROGRESSIVE_STICK_UPGRADE:
-            switch (CUR_UPG_VALUE(UPG_STICKS)) {
-                case 0:
-                case 1:
-                    return GI_STICK_UPGRADE_20;
-                case 2:
-                    return GI_STICK_UPGRADE_30;
-            }
-            return GI_RUPEE_BLUE;
-        
-        case RG_PROGRESSIVE_BOMBCHUS:
-            return GI_BOMBCHUS_20; //todo progressive?
-        
         case RG_PROGRESSIVE_MAGIC_METER:
             switch (gSaveContext.magicLevel) {
                 case 0:
                     return (GetItemID)RG_MAGIC_SINGLE;
                 case 1:
                     return (GetItemID)RG_MAGIC_DOUBLE;
             }
```

Comment:
ah, i think i see how the RG for single/double is working now, wouldn't that mean we'd never get the progressive item in the obtainability check?

Reply:
Yup, Progressive Item goes into the itemLocations table, and then these are returned to be pulled out of Randomizer's Get Item Table. IIRC this is basically how it was already working before I split out Randomizer's `GetItemEntry`s, just with custom `GetItemIDs` instead.